src/uct/ib/dc/dc_mlx5.c

```diff
@@ -324,7 +324,8 @@ static ucs_status_t uct_dc_mlx5_iface_create_dci(uct_dc_mlx5_iface_t *iface,
         attr.super.max_inl_cqe[UCT_IB_DIR_RX] = 0;
         attr.uidx             = htonl(dci_index) >> UCT_IB_UIDX_SHIFT;
         attr.full_handshake   = full_handshake;
-        attr.rdma_wr_disabled = iface->flags & UCT_DC_MLX5_IFACE_FLAG_DISABLE_PUT;
+        attr.rdma_wr_disabled = (iface->flags & UCT_DC_MLX5_IFACE_FLAG_DISABLE_PUT) &
+                                (md->flags & UCT_IB_MLX5_MD_FLAG_NO_RDMA_WR_OPTIMIZED);
```

Comment:
maybe add ucs_debug message to uct_ib_mlx5_devx_create_qp, that indicates which value we passed to rdma_wr_disabled

Reply:
Are you sure? In general I don't see such messages for other flags and fields in this file or in `ib_mlx5_dv.c`